### PR TITLE
Add epoch into avm tx verification and update create asset tx rules

### PR DESCRIPTION
**Pull request by aaronbuchwald**

### CODE REVIEW

File: vms/avm/create_asset_tx.go [hunk 1]
```diff
@@ -111,7 +134,60 @@ func (t *CreateAssetTx) SyntacticVerify(
...
+	if !utf8.ValidString(t.Name) {
+		return errInvalidNameApricot
+	}
+
+	for _, r := range t.Symbol {
+		if r > unicode.MaxASCII || r < 32 {
+			return errInvalidSymbolApricot
+		}
+	}
+
```
Comment (aaronbuchwald) [inline vms/avm/create_asset_tx.go:189]: Is this too permissive @akly ?

Comment (StephenButtolph) [reply vms/avm/create_asset_tx.go:189]: Can we exclude the max ascii value as well? `DEL` seems confusing and unnecessary.

`if r >= unicode.MaxASCII || r < 32 {`

File: snow/engine/avalanche/transitive.go [hunk 2]
```diff
@@ -466,12 +466,8 @@ func (t *Transitive) issue(vtx avalanche.Vertex, updatedEpoch bool) error {
...
 		for _, depID := range tr.Dependencies() {
 			if !trIDs.Contains(depID) && !t.Consensus.TransitionProcessing(depID) {
-				dep, err := t.VM.Get(depID)
-				if err != nil || !dep.Status().Decided() {
-					// This transaction hasn't been issued yet. Add it as a dependency.
-					t.missingTransitions.Add(depID)
-					i.txDeps.Add(depID)
-				}
+				t.missingTransitions.Add(depID)
+				i.txDeps.Add(depID)
```
Comment (danlaine) [inline snow/engine/avalanche/transitive.go:470]: Should we change `txDeps` to `trDeps`?

File: vms/avm/unique_tx.go [hunk 3]
```diff
@@ -57,11 +45,26 @@ type TxState struct {
 	status choices.Status
 }
 
+func newUniqueTx(vm *VM, txID ids.ID, rawTx *Tx) *UniqueTx {
```
Comment (danlaine) [inline vms/avm/unique_tx.go:48]: `// newUniqueTx returns the UniqueTx representation of transaction [txID]. rawTx may be nil.`

File: vms/avm/vm.go [hunk 4]
```diff
@@ -822,20 +812,21 @@ func (vm *VM) getUTXO(utxoID *avax.UTXOID) (*avax.UTXO, error) {
...
+	// If the parent is unknown, then it is too early to consume
+	// this UTXO.
 	if status := parent.Status(); status != choices.Processing {
 		return nil, errMissingUTXO
 	}
 
 	parentUTXOs := parent.UTXOs()
 	if uint32(len(parentUTXOs)) <= inputIndex || int(inputIndex) < 0 {
-		return nil, errInvalidUTXO
+		return nil, fmt.Errorf("invalid UTXO index %d for parent with %d UTXOs", inputIndex, len(parentUTXOs))
```
Comment (danlaine) [inline vms/avm/vm.go:829]: `fmt.Errorf("invalid output index %d for parent tx %s (%d outputs)", inputIndex, inputTxID, len(parentUTXOs))`

**Pull request merged**